xc/common/utils/prjxray_edge_library.py

```diff
@@ -1149,6 +1149,91 @@ def make_connection(
         yield edge
 
 
+def kill_track(conn, roi, synth_tracks):
+    """ Checks tracks with connections outside of the ROI.
+
+    Iterates over all alive tracks and removes the alive flag if
+    the track can lead to connections outside of the ROI.
+
+    Args:
+        conn (sqlite3.Connection): Connection database
+        roi: prjxray ROI object
+        synth_tracks : tracks related to synth tile connections (cannot be removed)
+
+    """
+
+    killed_tracks = set()
+    c = conn.cursor()
+    write_cur = conn.cursor()
+
+    for (track_pkey, ) in c.execute("""
+SELECT
+  pkey
+FROM
+  track
+WHERE
+  alive;"""):
+
+        write_cur.execute(
```

Comment:
Pull the constant source retrieval out of the loop?

Reply:
Done